src/Microsoft.Health.Fhir.Core/Features/Operations/Resources/Patch/PatchPayload.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.FhirPath;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Resources.Patch
+{
+    public abstract class PatchPayload
+    {
+        internal static ISet<string> ImmutableProperties =>
+            new HashSet<string>
+            {
+                "Resource.id",
+                "Resource.meta.lastUpdated",
+                "Resource.meta.versionId",
+                "Resource.text.div",
+                "Resource.text.status",
+            };
+
+        internal abstract ResourceElement GetPatchedResourceElement(ResourceWrapper resourceToPatch);
+
+        public ResourceElement Patch(ResourceWrapper resourceToPatch)
+        {
+            EnsureArg.IsNotNull(resourceToPatch, nameof(resourceToPatch));
+
+            // Capture the state of properties that are immutable
+            ITypedElement resource = resourceToPatch.RawResource.ToITypedElement(ModelInfoProvider.Instance);
```

Comment:
@LTA-Thinking - here I went from an `IModelInfoProvider` provided by the DI container to this statically provided version. Any issues with this approach?

I changed to this approach as `IModelInfoProvider` isn't injected in `FhirController`.

Reply:
This is the right way to do it. We use this pattern in other parts of our code.